api/imap-client/imap_client.py

```
@@ -82,16 +95,26 @@ def _messages_to_email_metadata(self, response):
 
         results = []
         for message_id, data in response.items():
-            formatted_data = self._get_formatted_message_data(
-                data)
+            # Process each message independently: a single malformed message
+            # must not abort the whole poll (which would drop every fetched
+            # message and re-raise to the Go caller as exit status 1).
+            try:
+                formatted_data = self._get_formatted_message_data(data)
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)